tests/unit/source_code/notebooks/test_cells.py

```diff
@@ -105,4 +107,33 @@ def test_pip_cell_build_dependency_graph_handles_multiline_code():
     problems = cell.build_dependency_graph(graph)
 
     assert len(problems) == 0
-    graph.register_library.assert_called_once_with("databricks")
+    graph.register_library.assert_called_once_with("databricks", installation_arguments=[])
+
+
+@pytest.mark.parametrize(
+    "code,split",
+    [
+        ("%pip install foo", ["%pip", "install", "foo"]),
+        ("%pip install", ["%pip", "install"]),
+        ("%pip installl foo", ["%pip", "installl", "foo"]),
+        ("%pip install foo --index-url bar", ["%pip", "install", "foo", "--index-url", "bar"]),
+        ("%pip install foo --index-url bar", ["%pip", "install", "foo", "--index-url", "bar"]),
+        ("%pip install foo --index-url \\\n bar", ["%pip", "install", "foo", "--index-url", "bar"]),
+        ("%pip install foo --index-url bar\nmore code", ["%pip", "install", "foo", "--index-url", "bar"]),
+        (
+            "%pip install foo --index-url bar\\\n -t /tmp/",
+            ["%pip", "install", "foo", "--index-url", "bar", "-t", "/tmp/"],
+        ),
+        ("%pip install foo --index-url \\\n bar", ["%pip", "install", "foo", "--index-url", "bar"]),
+        (
+            "%pip install ./distribution/dist/thingy-0.0.1-py2.py3-none-any.whl",
+            ["%pip", "install", "./distribution/dist/thingy-0.0.1-py2.py3-none-any.whl"],
+        ),
+        (
+            "%pip install distribution/dist/thingy-0.0.1-py2.py3-none-any.whl",
+            ["%pip", "install", "distribution/dist/thingy-0.0.1-py2.py3-none-any.whl"],
+        ),
+    ],
+)
+def test_pip_cell_split(code, split):
+    assert PipCell._split(code) == split  # pylint: disable=protected-access
```

Comment:
@nfx : We discussed this. Here the unit test for a protected method was useful to unity test many different pip install versions. It breaks the "no cheat" CI